docs/src/main/asciidoc/key-values.adoc

```diff
@@ -0,0 +1,101 @@
+== Key Values
+
+Azure App Configuration Supports multiple types of key values, some of which have special features built into them.
+
+=== Placeholders
+
+The client library supports configurations with `${}`-style environment placeholders. If referencing a Azure App Configuration key with a placeholder any prefix needs to be removed in the reference. For example: `/application/config.message` is referenced as `${config.message}`.
+
+NOTE: The prefix being removed matches the value `spring.cloud.azure.appconfiguration.stores[0].selects[0].key-filter`.
+
+=== JSON
+
+Configurations that have the a content-type `application/json` are processed as json objects. This enables one configuration mapping to a complex object in a `@Configuration`. The json key `/application/config.colors` with the value
+
+[source,json,indent=0]
+----
+{
+	"Red": {
+		"value": [255, 0, 0]
+	},
+	"Blue": {
+		"value": [0, 255, 0]
+	},
+	"Green": {
+		"value": [0, 0, 255]
+	}
+}
+----
+
+maps to
+
+[source,java,indent=0]
+----
+@ConfigurationProperties(prefix = "config")
+public class MyConfigurations {
+
+    private HashMap<String, Color> colors;
+
+}
+----
+
+=== Key Vault References
+
+Azure App Configuration and its client libraries support referencing secrets stored in Key Vault. In App Configuration keys can be created the have values that are URIs that map to secrets in a Key Vault. This enables using secrets as configuration without storing them as configurations.
```

Comment:
Second sentence is not clear. What about this: 
" In App Configuration, keys can be created which have URI values that map to a secret stored in a Key Vault. The secrets are securely stored in Key Vault, but can be accessed with the same syntax as configuration"